server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1956,16 +1952,16 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
 
             if (answer instanceof GetVolumeStatsAnswer){
                 GetVolumeStatsAnswer volstats = (GetVolumeStatsAnswer)answer;
-                return volstats.getVolumeStats();
+                volumeStatsByUuid.putAll(volstats.getVolumeStats());
             }
         }
-        return null;
+        return volumeStatsByUuid.size() > 0 ? volumeStatsByUuid : null;
     }
 
     private List<String> getVolumesByHost(HostVO host, StoragePool pool){
-        List<UserVmVO> vmsPerHost = _vmDao.listByHostId(host.getId());
+        List<VMInstanceVO> vmsPerHost = _vmInstanceDao.listByHostId(host.getId());
         return vmsPerHost.stream()
-                .flatMap(vm -> _volsDao.findByInstanceIdAndPoolId(vm.getId(),pool.getId()).stream().map(vol -> vol.getPath()))
+                .flatMap(vm -> _volsDao.findByInstanceIdAndPoolId(vm.getId(),pool.getId()).stream().map(vol -> vol.getState() == Volume.State.Ready ? vol.getPath() : null).filter(Objects::nonNull))
```

Comment:
@slavkap volume path is not the locator for OVA format, update with chain info ?

Reply:
@sureshanaparti I was just follow the original logic and just added to get only the volumes with status "Ready". I will check for the rest of the formats, but I will be not able to do tests for them

Reply:
ok @slavkap  you have corrected the volume locator for QCOW2 in the response helper, this would be similar change for OVA here.

Reply:
yes, I will add to check if it's OVA